[FEATURE] TYPO3 v12 Compatibility

Riiiad opened the pull request.

calien666 committed:
[TASK] Finalize events and code-style-fix

calien666 committed:
[TASK] Finalize events and code-style-fix

Riiiad committed:
[TASK] Add setup skip for LTS 12

Riiiad committed:
[TASK] Add setup skip for LTS 12

calien666 committed:
[TASK] Adjust composer for TYPO3 12

Used commands:
```shell
Build/Scripts/runTests.sh -s composer -- config sort-packages true
Build/Scripts/runTests.sh -s composer -- require \
    typo3/cms-core:^12.4 \
    typo3/cms-extbase:^12.4 \
    typo3/cms-fluid-styled-content:^12.4 \
    --no-update
Build/Scripts/runTests.sh -s composer -- require --dev \
    typo3/cms-felogin:^12.4 \
    --no-update
Build/Scripts/runTests.sh -s composer -- remove --dev \
    typo3/cms-tstemplate \
    --no-update
Build/Scripts/runTests.sh -s composer -- require --dev \
    typo3/testing-framework:^8.2 \
    --no-update
Build/Scripts/runTests.sh -s composer -- require --dev \
    friendsofphp/php-cs-fixer:^3.86 \
    --no-update
Build/Scripts/runTests.sh -s composer -- config --unset scripts
Build/Scripts/runTests.sh -s composer -- require \
    php:"^8.1 || ^8.2 || ^8.3 || ^8.4" \
    --no-update
Build/Scripts/runTests.sh -s composer -- require --dev \
    typo3/cms-install:^12.4 \
    --with-all-dependencies
```

calien666 committed:
[TASK] Adjust composer for TYPO3 12

Used commands:
```shell
Build/Scripts/runTests.sh -s composer -- config sort-packages true
Build/Scripts/runTests.sh -s composer -- require \
    typo3/cms-core:^12.4 \
    typo3/cms-extbase:^12.4 \
    typo3/cms-fluid-styled-content:^12.4 \
    --no-update
Build/Scripts/runTests.sh -s composer -- require --dev \
    typo3/cms-felogin:^12.4 \
    --no-update
Build/Scripts/runTests.sh -s composer -- remove --dev \
    typo3/cms-tstemplate \
    --no-update
Build/Scripts/runTests.sh -s composer -- require --dev \
    typo3/testing-framework:^8.2 \
    --no-update
Build/Scripts/runTests.sh -s composer -- require --dev \
    friendsofphp/php-cs-fixer:^3.86 \
    --no-update
Build/Scripts/runTests.sh -s composer -- config --unset scripts
Build/Scripts/runTests.sh -s composer -- require \
    php:"^8.1 || ^8.2 || ^8.3 || ^8.4" \
    --no-update
Build/Scripts/runTests.sh -s composer -- require --dev \
    typo3/cms-install:^12.4 \
    --with-all-dependencies
```

calien666 committed:
[TASK] Streamline code quality tools

* Adjust runTests.sh to newer version and drop docker-compose.yml
* Adjust phpstan
* Unify unit testing
* correct cs fixer
* remove last ddev file

calien666 committed:
[TASK] Streamline code quality tools

* Adjust runTests.sh to newer version and drop docker-compose.yml
* Adjust phpstan
* Unify unit testing
* correct cs fixer
* remove last ddev file

calien666 committed:
[TASK] Free ViewHelpers from deprecated `CompileWithRenderStatic`

calien666 committed:
[TASK] Free ViewHelpers from deprecated `CompileWithRenderStatic`

calien666 committed:
[WIP][TASK] Code updates

* Resolve deprecations and breakings

calien666 committed:
[WIP][TASK] Code updates

* Resolve deprecations and breakings

calien666 committed:
[RECTOR] Run ReadOnlyPropertyRector

calien666 committed:
[RECTOR] Run ReadOnlyPropertyRector

calien666 committed:
[RECTOR] Run ClassPropertyAssignToConstructorPromotionRector

calien666 committed:
[RECTOR] Run ClassPropertyAssignToConstructorPromotionRector

calien666 committed:
[FRACTOR] Run RemoveTceFormsDomElementFlexFormFractor

calien666 committed:
[FRACTOR] Run RemoveTceFormsDomElementFlexFormFractor

calien666 committed:
[RECTOR] Run DisallowedEmptyRuleFixerRector

calien666 committed:
[RECTOR] Run DisallowedEmptyRuleFixerRector

calien666 committed:
[RECTOR] Run NullToStrictStringFuncCallArgRector

calien666 committed:
[RECTOR] Run NullToStrictStringFuncCallArgRector

calien666 committed:
[RECTOR] Run ExtbaseAnnotationToAttributeRector

calien666 committed:
[RECTOR] Run ExtbaseAnnotationToAttributeRector

calien666 committed:
[RECTOR] Run CombinedAssignRector

calien666 committed:
[RECTOR] Run CombinedAssignRector

calien666 committed:
[RECTOR] Run RenameClassConstFetchRector

calien666 committed:
[RECTOR] Run RenameClassConstFetchRector

calien666 committed:
[RECTOR] Run ExplicitBoolCompareRector

calien666 committed:
[RECTOR] Run ExplicitBoolCompareRector

calien666 committed:
[RECTOR] Run ExtbaseAnnotationToAttributeRector

calien666 committed:
[RECTOR] Run ExtbaseAnnotationToAttributeRector

calien666 committed:
[RECTOR] Run CombineIfRector

calien666 committed:
[RECTOR] Run CombineIfRector

calien666 committed:
[RECTOR] Run SafeDeclareStrictTypesRector

calien666 committed:
[RECTOR] Run SafeDeclareStrictTypesRector

calien666 committed:
[RECTOR] Run MoveAllowTableOnStandardPagesToTCAConfigurationRector

calien666 committed:
[RECTOR] Run MoveAllowTableOnStandardPagesToTCAConfigurationRector

calien666 committed:
[RECTOR] Run RemoveUnusedVariableInCatchRector

calien666 committed:
[RECTOR] Run RemoveUnusedVariableInCatchRector

calien666 committed:
[RECTOR] Run MigrateItemsIndexedKeysToAssociativeRector

calien666 committed:
[RECTOR] Run MigrateItemsIndexedKeysToAssociativeRector

calien666 committed:
[RECTOR] Run MigrateToEmailTypeRector

calien666 committed:
[RECTOR] Run MigrateToEmailTypeRector

calien666 committed:
[RECTOR] Run RemoveCruserIdRector

calien666 committed:
[RECTOR] Run RemoveCruserIdRector

calien666 committed:
[RECTOR] Run MigrateEvalIntAndDouble2ToTypeNumberRector

calien666 committed:
[RECTOR] Run MigrateEvalIntAndDouble2ToTypeNumberRector

calien666 committed:
[RECTOR] Run FlipTypeControlToUseExclusiveTypeRector

calien666 committed:
[RECTOR] Run FlipTypeControlToUseExclusiveTypeRector

calien666 committed:
[RECTOR] Run MigrateInputDateTimeRector

calien666 committed:
[RECTOR] Run MigrateInputDateTimeRector

calien666 committed:
[RECTOR] Run CompleteDynamicPropertiesRector

calien666 committed:
[RECTOR] Run CompleteDynamicPropertiesRector

calien666 committed:
[RECTOR] Run ClassOnObjectRector

calien666 committed:
[RECTOR] Run ClassOnObjectRector

calien666 committed:
[RECTOR] Run MigrateRequiredFlagRector

calien666 committed:
[RECTOR] Run MigrateRequiredFlagRector

calien666 committed:
[RECTOR] Run ReadOnlyPropertyRector

calien666 committed:
[RECTOR] Run ReadOnlyPropertyRector

calien666 committed:
[RECTOR] Run ClassConstantToSelfClassRector

calien666 committed:
[RECTOR] Run ClassConstantToSelfClassRector

calien666 committed:
[RECTOR] Run RenameClassConstFetchRector

calien666 committed:
[RECTOR] Run RenameClassConstFetchRector

calien666 committed:
[RECTOR] Run ShortenElseIfRector

calien666 committed:
[RECTOR] Run ShortenElseIfRector

calien666 committed:
[RECTOR] Run SimplifyIfElseToTernaryRector

calien666 committed:
[RECTOR] Run SimplifyIfElseToTernaryRector

calien666 committed:
[RECTOR] Run IfIssetToCoalescingRector

calien666 committed:
[RECTOR] Run IfIssetToCoalescingRector

calien666 committed:
[RECTOR] Run GeneralUtilityMakeInstanceToConstructorPropertyRector

calien666 committed:
[RECTOR] Run GeneralUtilityMakeInstanceToConstructorPropertyRector

calien666 committed:
[RECTOR] Run SetCookieRector

calien666 committed:
[RECTOR] Run SetCookieRector

calien666 committed:
[RECTOR] Run InjectMethodToConstructorInjectionRector

calien666 committed:
[RECTOR] Run InjectMethodToConstructorInjectionRector

calien666 committed:
[RECTOR] Run ClassPropertyAssignToConstructorPromotionRector

calien666 committed:
[RECTOR] Run ClassPropertyAssignToConstructorPromotionRector

calien666 committed:
[RECTOR] Run ExtEmConfRector

calien666 committed:
[RECTOR] Run ExtEmConfRector

calien666 committed:
[TASK] CGL run

Used command:
```
Build/Scripts/runTests.sh -s cgl
```

calien666 committed:
[TASK] CGL run

Used command:
```
Build/Scripts/runTests.sh -s cgl
```

calien666 committed:
[TASK] remove strict type declaration from ext_emconf.php

When not fitting the TER given structure, the ext_emconf is overridden
and will not been set by the extension itself.

calien666 committed:
[TASK] remove strict type declaration from ext_emconf.php

When not fitting the TER given structure, the ext_emconf is overridden
and will not been set by the extension itself.

calien666 committed:
[TASK] Resolve phpstan issues

calien666 committed:
[TASK] Resolve phpstan issues

calien666 committed:
[TASK] Change tests to core 12

calien666 committed:
[TASK] Change tests to core 12

calien666 committed:
[TASK] Add TypoScript Linter

used command:
```
Build/Scripts/runTests.sh -s composer -- require --dev helmich/typo3-typoscript-lint:^3.3
```

calien666 committed:
[TASK] Add TypoScript Linter

used command:
```
Build/Scripts/runTests.sh -s composer -- require --dev helmich/typo3-typoscript-lint:^3.3
```

calien666 committed:
[TASK] Fix deprecations detected by functional tests

calien666 committed:
[TASK] Fix deprecations detected by functional tests

coding46 committed:
[TASK] Fix FlexForm localization

coding46 commented on Classes/Controller/PollController.php:
```diff
@@ -1097,7 +1076,7 @@ private function processPollAndVoteArgumentFromRequest(): void
                 $poll['options'] = array_values($poll['options']);
             }
-            $this->request->setArgument('poll', $poll);
+            $this->arguments->addNewArgument('poll', 'Array', false, $poll);
```
`addNewArgument('poll', ...)` replaces the Extbase argument definition, not the incoming request argument. In TYPO3 12, Extbase still maps later from
  `request->getArgument('poll')`, so the sanitized poll data prepared here is very likely ignored.

coding46 replied:
Fixed

coding46 commented on ext_localconf.php:
```diff
@@ -54,15 +51,4 @@
-    $GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['ext/install']['update']['t3oodleMigrateOneOptionOnlySetting']
-        = MigrateOneOptionOnlySetting::class;
-    $GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['ext/install']['update']['t3oodleMigrateOldPollTypes']
-        = MigrateOldPollTypes::class;
```
This removes the old upgrade wizards without adding a replacement migration path. If upgrading existing t3oodle installations is still supported, older
  persisted poll data may no longer be migrated correctly on TYPO3 12.

coding46 replied:
The Removal is correct as the support for two TYPO3 major versions does not include v9 and v10. The wizards were introduced for those versions back then.

coding46 committed:
[TASK] Initialize user ident safely

coding46 committed:
[TASK] Fix poll repository object type initialization

Ensures the repository keeps BasePoll::class as its managed object type
after the parent repository constructor runs.

Without this, Extbase derives a non-existent
FGTCLB\T3oodle\Domain\Model\Poll model class from the repository name
and query creation fails.

coding46 committed:
[TASK] Fix SVG parsing in view helper

Adjusts the SVG parser to handle DOMDocument::loadXML() correctly in the
current runtime.

The view helper now checks the boolean parse result and continues
working with the populated DOMDocument instance instead of treating the
return value itself as a document object.

coding46 committed:
[TASK] Use request argument mutation for poll preprocessing

Replaces addNewArgument() with withArgument() when normalizing submitted
poll data.

This keeps the original Extbase argument definition intact and updates
the actual request payload that is later used for property mapping.

coding46 committed:
[TASK] Initialize vote option values safely

Ensures transient Vote objects always expose an initialized optionValues
storage during rendering and form mapping.

coding46 committed:
[TASK] Map poll frontend users to fe_users

Add the missing Extbase persistence mapping for PollFrontendUser so
frontend user lookups use fe_users instead of a non-existent extension
table.

coding46 committed:
[TASK] Tighten DOM type checks in SVG view helper

Add explicit DOM root and XML serialization guards so the SVG view
helper satisfies static analysis under PHP 8.1 without changing its
runtime behavior.

coding46 committed:
[TASK] Fix create action mapping for poll subtypes

Pass the submitted poll subtype through Extbase property mapping in
initializeCreateAction() so createAction() receives a concrete poll
model instead of trying to instantiate abstract BasePoll.

coding46 committed:
[TASK] Fix frontend new poll permission handling

Allow subject-less new poll permission checks in Fluid and guard newAction()
against requests that do not yet carry a poll argument.

This restores rendering of the new poll buttons and prevents the initial
new poll request from failing before the form is shown.

coding46 committed:
[TASK] Fix guest poll publish permission after create

Refresh the current poll permission state after generating a guest author
ident during poll creation.

This keeps the immediate create-to-publish forward consistent so newly
created guest polls can be published right away.

sbuerk committed:
[TASK] Set dev-version `2.0.0-dev`

Used command(s):

```bash
echo '>> Create release' ; \
  RELEASE_BRANCH='master' ; \
  RELEASE_VERSION='1.0.1' ; \
  DEV_VERSION='2.0.0' ; \
  echo ">> Post-release - set dev version: ${DEV_VRESION}-dev" && \
  sed -i "s/^COMPOSER_ROOT_VERSION.*/COMPOSER_ROOT_VERSION=\"${DEV_VERSION}-dev\"/" Build/Scripts/runTests.sh && \
  tailor set-version ${DEV_VERSION} && \
  composer config "extra"."typo3/cms"."version" "${DEV_VERSION}-dev" && \
  echo "${DEV_VERSION}-dev" > VERSION
```
